src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -3117,6 +3286,296 @@ NetworkOPsImp::unsubAccountInternal(
+                                inner)
+                            {
+                                if (inner->getAccountID(sfAccount) == accountId)
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                    }
+                    catch (std::exception const&)
+                    {
+                        continue;
+                    }
+                }
+
+                return false;
+            };
+
+            auto send = [&](Json::Value const& jvObj,
+                            bool unsubscribe) -> bool {
+                if (auto sptr = subInfo.sinkWptr_.lock(); sptr)
+                {
+                    sptr->send(jvObj, true);
+                    if (unsubscribe)
+                        unsubAccountHistory(sptr, accountId, false);
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            };
+
+            /*
+             * search backward until the genesis ledger or asked to stop
+             */
+            while (lastLedgerSeq >= 2 && !subInfo.index_->stopHistorical_)
+            {
+                auto startLedgerSeq =
+                    (lastLedgerSeq > 256 + 2 ? lastLedgerSeq - 256 : 2);
+                JLOG(m_journal.trace())
+                    << "AccountHistory job for account " << toBase58(accountId)
+                    << ", working on ledger range [" << startLedgerSeq << ","
+                    << lastLedgerSeq << "]";
+
+                auto haveRange = [&]() -> bool {
+                    std::uint32_t validatedMin = UINT_MAX;
+                    std::uint32_t validatedMax = 0;
+                    auto haveSomeValidatedLedgers =
+                        app_.getLedgerMaster().getValidatedRange(
+                            validatedMin, validatedMax);
+
+                    return haveSomeValidatedLedgers &&
+                        validatedMin <= startLedgerSeq &&
+                        lastLedgerSeq <= validatedMax;
+                }();
+
+                if (!haveRange)
+                {
+                    JLOG(m_journal.debug())
+                        << "AccountHistory reschedule job for account "
+                        << toBase58(accountId) << ", incomplete ledger range ["
+                        << startLedgerSeq << "," << lastLedgerSeq << "]";
+                    setAccountHistoryJobTimer(subInfo);
+                    return;
+                }
+
+                RelationalDBInterface::AccountTxPageOptions options = {
+                    accountId, startLedgerSeq, lastLedgerSeq, {}, 0, true};
+
+                auto [txns, marker] =
+                    dynamic_cast<RelationalDBInterfaceSqlite*>(
+                        &app_.getRelationalDBInterface())
+                        ->newestAccountTxPage(options);
```

Comment:
I think we should charge the consumer for each of these calls. These calls are heavy weight, and impose a lot of load on the server. Here is an example of how to do that: https://github.com/ripple/rippled/blob/8a25f3282426f382e1933242ef875d5aaf3e31e9/src/ripple/rpc/impl/ServerHandlerImp.cpp#L425
We usually consider account_tx calls as `feeMediumBurdenRPC`

Reply:
Thanks, good idea, added.